src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -198,6 +198,56 @@ private static async Task QueueBuildPolicy(GitPullRequest pullRequest, string bu
             }
         }
 
+        /// <summary>
+        /// There is no enum or class in Microsoft.TeamFoundation.SourceControl.WebApi defined for vote values so made my own here.
+        /// Values are documented at https://www.visualstudio.com/en-us/docs/integrate/api/git/pull-requests/reviewers#update-a-reviewers-vote.
+        /// </summary>
+        public enum Vote : short
+        {
+            Approved = 10,
+            ApprovedWithComment = 5,
+            NoResponse = 0,
+            NotReady = -5,
+            Rejected = -10
+        }
+
+        // Similar to: https://devdiv.visualstudio.com/DevDiv/_git/PostBuildSteps#path=%2Fsrc%2FSubmitPullRequest%2FProgram.cs&version=GBmaster&_a=contents
+        private static async Task SetAutoCompleteAsync(GitPullRequest pullRequest)
+        {
+            var gitClient = ProjectCollection.GetClient<GitHttpClient>();
+            var repository = pullRequest.Repository;
+            try
+            {
+                var idRefWithVote = await gitClient.CreatePullRequestReviewerAsync(
+                    new IdentityRefWithVote { Vote = (short)Vote.Approved },
+                    repository.Id,
+                    pullRequest.PullRequestId,
+                    VSLSnapUserId.ToString()
+                    );
+                Console.WriteLine($"Updated {pullRequest.Description} with AutoApprove");
+
+                pullRequest = await gitClient.UpdatePullRequestAsync(
+                    new GitPullRequest
+                    {
+                        AutoCompleteSetBy = idRefWithVote,
+                        CompletionOptions = new GitPullRequestCompletionOptions
+                        {
+                            DeleteSourceBranch = true,
+                            MergeCommitMessage = "RIT Auto-Complete",
+                            SquashMerge = true,
+                        }
+                    },
+                    repository.Id,
+                    pullRequest.PullRequestId
+                    );
+                Console.WriteLine($"Updated {pullRequest.Description} with AutoComplete");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Could not set AutoComplete: {e.GetType().Name} : {e.Message}");
```

Comment:
Not sure if I should throw here. Don't want the release to fail if we couldn't set the PR to auto-complete but also want to inform the user - is it best to throw or not?

Reply:
WE typically logged things here. I think all that code was removed. I would at least write to the console window so people know what happened

---
In reply to: [377406314](https://github.com/dotnet/roslyn-tools/pull/698#discussion_r377406314) [](ancestors = 377406314)

Reply:
There should be a LogWarning method you can use which will make it show up as a warning in the release log. e.g. https://devdiv.visualstudio.com/DevDiv/_releaseProgress?_a=release-environment-logs&releaseId=580152&environmentId=3097439